### PR TITLE
Seed script

### LINKED ISSUE #325: Create a seed script
Out-of-the-box Shifts is useless and hard to test. We should create a basic seed script that will populate the database with a location and users with multiple roles.

### PR DESCRIPTION
**Pull request by njlxyaoxinwei**
Fixes #325 
Now running `rake db:reset` will automatically populate a dummy database! No need for the anonymized DB anymore

### CODE REVIEW

File: db/seeds.rb [hunk 1]
```diff
@@ -5,3 +5,468 @@
...
+Active_notice_chance = 0.8
+Public_loc_group_chance = 0.6
+User_with_nick_name_chance = 0.3
+User_profile_complete_chance = 0.9
+User_has_pic_chance = 0.4
+Location_has_time_slot_chance = 0.8
+Weekday_has_time_slot_chance = 5/7.0
+Shift_signed_in_chance = 0.9
+Payform_submitted_chance = 0.9
+Payform_printed_chance = 0.7
```
Comment (mnquintana) [inline db/seeds.rb]: Why are all these variables capitalized? That's usually reserved for class names.

File: db/seeds.rb [hunk 2]
```diff
@@ -5,3 +5,468 @@
...
+
+def random_user
+  rand_id = rand(User.count)+1
+  User.where("id > ?", rand_id).first
+end
+################################################################################
+## Record Generating Methods
+################################################################################
+def app_config_gen
+	ap = AppConfig.new
```
Comment (mnquintana) [inline db/seeds.rb]: Wonky indenting

File: db/seeds.rb [hunk 3]
```diff
@@ -5,3 +5,468 @@
...
+  ap.mailer_address = "shifts.app@yale.edu"
+  ap.admin_email = Faker::Internet.email
+  ap.use_ldap = false
+  ap.calendar_feed_hash = SecureRandom.hex(32)
+  ap.save!
+  ap
+end
+
+def user_gen
+	user = User.new
```
Comment (mnquintana) [inline db/seeds.rb]: The indentation in this method is all over the place

File: db/seeds.rb [hunk 4]
```diff
@@ -5,3 +5,468 @@
...
+end
+
+################################################################################
+## Beginning of seed script
+################################################################################
+Timecop.travel(DateTime.now - Ndays_of_schedule_into_past.days)
+
+# First AppConfig
+progress_bar_gen("AppConfig [1/15]", 1) do
+  @app_config = app_config_gen
```
Comment (mnquintana) [inline db/seeds.rb]: Why is this (and some later vars) an instance variable?

File: db/seeds.rb [hunk 5]
```diff
@@ -5,3 +5,468 @@
...
+  Category.all.each {|cat| cat.update_attributes(billing_code: ptaeo_gen)}
+  @category = @department.categories.first
+  @calendar = @department.calendars.default
+end
+
+# Create superuser
+progress_bar_gen("Superuser [3/15]", 1) do 
+  @su = user_gen
+  @su.superuser = true
+  puts;prompt_field(@su, "login")
```
Comment (mnquintana) [inline db/seeds.rb]: Any reason why this needs to be one line?

File: db/seeds.rb [hunk 6]
```diff
@@ -5,3 +5,468 @@
...
+# Creating Payforms
+progress_bar_gen("Payforms [15/15]", User.count*NPayform_item_per_user) do |bar|
+  categories = Category.all
+  User.all.each do |user|
+    payform = Payform.build(@department, user, Date.today)
+    NPayform_item_per_user.times do 
+      payform_item_gen(payform, categories.sample)
+      bar.increment
+    end
+    if rand()<Payform_submitted_chance and user!=@su
```
Comment (mnquintana) [inline db/seeds.rb]: Usually `&&` is preferred over `and` in Ruby

File: db/seeds.rb [hunk 7]
```diff
@@ -5,3 +5,468 @@
...
+    puts e.message
+  end
+end
+
+def shift_gen(location, date, user)
+  shift = Shift.new(calendar: @calendar, department: @department, location: location, user: user)
+  start_hour = @department_config.schedule_start/60
+  end_hour = @department_config.schedule_end/60
+  shift_start = date + (start_hour+rand(end_hour-start_hour)).hours
+  duration = Shifts_duration_min_hours+rand(Shifts_duration_max_hours-Shifts_duration_min_hours+1)
```
Comment (mnquintana) [inline db/seeds.rb]: Can you put spaces around all operators?

File: db/seeds.rb [hunk 8]
```diff
@@ -5,3 +5,468 @@
...
+  field3 = @department.user_profile_fields.create!(name: "Class", display_type: "select", public: true, user_editable: true, index_display: true, values: "2014,2015,2016,2017,2018,Other")
+  # Multiple Choice
+  field4 = @department.user_profile_fields.create!(name: "Gender", display_type: "radio_button", public: false, user_editable: false, index_display: true, values: "M,F")
+  # Check boxes
+  field5 = @department.user_profile_fields.create!(name: "OS", display_type: "check_box", public: true, user_editable: true, index_display: false, values: "Linux, OS X, Windows")
+  # Picture Link
+  field6 = @department.user_profile_fields.create!(name: "Pic", display_type: "picture_link", public: true, user_editable: true, index_display: false)
+end
+
+
```
Comment (mnquintana) [inline db/seeds.rb]: Extra newline here

File: db/seeds.rb [hunk 9]
```diff
@@ -5,3 +5,468 @@
...
+  end until User.where(login: login).empty?
+  user.login = login
+  user.set_random_password
+  user.departments = [@department]
+  user.save!
+  user.set_payrate(10+rand(10), @department)
+  user
+end
+
+
```
Comment (mnquintana) [inline db/seeds.rb]: Too many newlines

**Comment by mnquintana** [comment]
Sorry to have ended up being code-style police :no_mouth: but otherwise awesome work! (I haven't tried running it yet though.)

File: db/seeds.rb (hunk 1 above)
Comment (njlxyaoxinwei) [inline db/seeds.rb]: I am not sure about this either. Reservations seed script uses ALL CAPS for the parameters, but I have very long variable name here so I only capitalized first letter. The idea is to make the variable accessible in the whole script, including inside methods.

File: db/seeds.rb (hunk 4 above)
Comment (njlxyaoxinwei) [inline db/seeds.rb]: To make it accessible in the entire script, including when inside methods. As this is the practice both within the app (controller/view) and within the test (helper/spec) for these ActiveRecord variables, I have adopted the same thing here in the seed script

File: db/seeds.rb (hunk 5 above)
Comment (njlxyaoxinwei) [inline db/seeds.rb]: Without a newline the output is weird, I put them on the same line to suggest that both are part of one command-line I/O action.

**Pull request merged**